modules/standard/IO.chpl

```diff
@@ -2860,6 +2868,9 @@ private inline proc _read_binary_internal(_channel_internal:qio_channel_ptr_t, p
                                       _channel_internal, tx, len, -1);
     x = new string(tx, length=len, needToCopy=false);
     return ret;
+  } else if t == _bytes {
+    // handle _bytes
+    halt("Cannot read into bytes objects yet");
```

Comment:
This case is easy in the binary channel, isn't it? AFAIK qio_channel_read_string doesn't assume anything about UTF-8. Nonetheless it's OK to do it in another PR.